lib/pages/timetable/service/uni_event_provider.dart

```diff
@@ -234,14 +263,20 @@ class UniEventProvider extends EventProvider<UniEventInstance>
           try {
             for (final doc in snapshot.docs) {
               ClassHeader classHeader;
-              final data = doc.data();
-              if (data['class'] != null) {
+              Person teacher;
+              if (doc.data()['class'] != null) {
```

Comment:
Save doc.data() in a variable (in AS, select doc.data() and click ctrl+alt+v and it should automatically replace all occurrences and extract a variable.